Remove dependency on external library/grape

robsyme opened the pull request:
Switching to vanilla Groovy for API interaction to remove requirement for the workflow to reach out to an external service (Maven) to pull extra libraries.

robsyme committed:
Remove dependency on external library/grape

robsyme committed:
Clarify error message when fetching organization details

robsyme committed:
Follow best-practices from Nextflow LSP

drpatelh commented on modules/local/seqera_runs_dump/main.nf:
```diff
@@ -18,11 +18,11 @@ process SEQERA_RUNS_DUMP {
-    fusion = metaOut.fusion ? '--add-fusion-logs' : ''
-    javaTrustStore = java_truststore_path ? "-Djavax.net.ssl.trustStore=${java_truststore_path}" : ''
-    javaTrustStorePassword = java_truststore_password ? "-Djavax.net.ssl.trustStorePassword=${java_truststore_password}" : ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def metaOut = meta + getRunMetadata(meta, log, api_endpoint, java_truststore_path, java_truststore_password)
```
```suggestion
    prefix = task.ext.prefix ?: "${meta.id}"
    metaOut = meta + getRunMetadata(meta, log, api_endpoint, java_truststore_path, java_truststore_password)
```

drpatelh replied:
These variables are used in the `output` section so we can't scope with `def`? Be good to test this if we haven't already.

robsyme replied:
Why remove the `def`s here, Harshil? The LSP recommends adding them. Do you think adding in the def statements will break something?

robsyme replied:
Ah, ok. Let me test.

robsyme commented:
Note - this is a PR to dev. If we'd prefer that this go to `master` for a patch release, I'm ok with that.

drpatelh commented:
Let's PR to dev as we normally do. 
- We need to bump versions etc too. 
- Wonder if we should add a config for `maxForks` by default too. We have seen this happen quite a bit due to API limits.

robsyme committed:
Update modules/local/seqera_runs_dump/main.nf

Harshil's suggestion is correct. Even though LSP recommends the defs be added, this would break the process.

Co-authored-by: Harshil Patel <drpatelh@users.noreply.github.com>

robsyme committed:
Don't believe LSP

robsyme committed:
Fix ordering
